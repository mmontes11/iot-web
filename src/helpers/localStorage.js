const languageKey = "language";

export const setLanguage = language => window.localStorage.setItem(languageKey, language);
export const getLanguage = () => window.localStorage.getItem(languageKey);
