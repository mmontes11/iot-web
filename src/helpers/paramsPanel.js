export const isObservationDisabled = observation => {
  if (observation && observation.isDisabled !== undefined) {
    return observation.isDisabled;
  }
  return true;
};
