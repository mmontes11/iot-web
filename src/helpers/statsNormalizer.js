export const normalizeStats = stats => {
  const keys = Object.keys(stats);
  const itemKeys = keys.filter(key => key !== "data");
  const items = itemKeys.map(itemKey => ({
    name: itemKey,
    value: stats[itemKey],
  }));
  return {
    data: items,
    title: stats.data,
  };
};
