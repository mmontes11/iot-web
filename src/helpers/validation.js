export const isNullOrEmpty = array => array === null || array.length === 0;
