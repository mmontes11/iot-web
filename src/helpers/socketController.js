import SocketIOClient from "socket.io-client";
import { getToken } from "helpers/localStorage";

export class SocketController {
  constructor(thing, onData) {
    const query = {
      token: getToken(),
      thing,
    };
    this.socket = new SocketIOClient(process.env.IOT_SERVER_URL, { query });
    this.onData = onData;
  }
  listen() {
    this.socket.on("data", data => this.onData(data));
  }
  close() {
    this.socket.close();
  }
}
