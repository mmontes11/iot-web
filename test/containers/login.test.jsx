import React from "react";
import { mount } from "enzyme";
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import configureStore from "redux-mock-store";
import Login from "containers/login";
import { initialState } from "../helpers/redux";

describe("containers/login", () => {
  it("renders login in initial state", () => {
    const store = configureStore([thunk])(initialState);
    const wrapper = mount(
      <Provider store={store}>
        <Login />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("renders login in loading state", () => {
    const state = {
      ...initialState,
      auth: {
        isAuth: true,
        username: "username",
        password: "password",
        showError: true,
      },
      request: {
        pending: 1,
        statusCode: null,
        error: null,
      },
    };
    const store = configureStore([thunk])(state);
    const wrapper = mount(
      <Provider store={store}>
        <Login />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("renders login in error state", () => {
    const state = {
      ...initialState,
      auth: {
        isAuth: true,
        username: "username",
        password: "password",
        showError: true,
      },
      request: {
        pending: 1,
        statusCode: null,
        error: new Error(),
      },
    };
    const store = configureStore([thunk])(state);
    const wrapper = mount(
      <Provider store={store}>
        <Login />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("simulates closing dialog", () => {
    const store = configureStore([thunk])(initialState);
    const wrapper = mount(
      <Provider store={store}>
        <Login />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
    wrapper.find(".modal-close").simulate("click");
    expect(store.getState()).toMatchSnapshot();
    expect(wrapper).toMatchSnapshot();
  });
  it("simulates successful login", () => {
    const store = configureStore([thunk])(initialState);
    const wrapper = mount(
      <Provider store={store}>
        <Login />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
    wrapper.find("#username-input").simulate("change", { target: { value: "username" } });
    wrapper.find("#password-input").simulate("change", { target: { value: "password" } });
    wrapper.find("#login-button").simulate("click");
    expect(store.getState()).toMatchSnapshot();
    expect(wrapper).toMatchSnapshot();
  });
});
