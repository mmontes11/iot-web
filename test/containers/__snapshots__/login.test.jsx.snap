// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/login renders login in initial state and unmounts 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Connect(ResetOnUnmount))>
    <Connect(ResetOnUnmount)
      isAuth={[Function]}
      isLoading={false}
      login={[Function]}
      logout={[Function]}
      password={null}
      setPassword={[Function]}
      setUsername={[Function]}
      username={null}
    >
      <ResetOnUnmount
        isAuth={[Function]}
        isLoading={false}
        login={[Function]}
        logout={[Function]}
        password={null}
        reset={[Function]}
        setPassword={[Function]}
        setUsername={[Function]}
        username={null}
      >
        <Login
          isAuth={[Function]}
          isLoading={false}
          login={[Function]}
          logout={[Function]}
          password={null}
          reset={[Function]}
          setPassword={[Function]}
          setUsername={[Function]}
          username={null}
        >
          <section
            className="hero is-fullheight"
          >
            <div
              className="hero-body"
            >
              <div
                className="container has-text-centered"
              >
                <div
                  className="column is-4 is-offset-4"
                >
                  <div
                    className="box"
                  >
                    <p
                      className="title is-1"
                    >
                      IoT
                    </p>
                    <form>
                      <div
                        className="field"
                      >
                        <div
                          className="control has-icons-left"
                        >
                          <input
                            className="input is-large"
                            id="username-input"
                            onChange={[Function]}
                            placeholder="Username"
                            value=""
                          />
                          <span
                            className="icon is-large is-left"
                          >
                            <i
                              className="fas fa-user"
                            />
                          </span>
                        </div>
                      </div>
                      <div
                        className="field"
                      >
                        <div
                          className="control has-icons-left"
                        >
                          <input
                            className="input is-large"
                            id="password-input"
                            onChange={[Function]}
                            placeholder="Password"
                            type="password"
                            value=""
                          />
                          <span
                            className="icon is-large is-left"
                          >
                            <i
                              className="fas fa-lock"
                            />
                          </span>
                        </div>
                      </div>
                      <button
                        className="button is-block is-primary is-large is-fullwidth"
                        disabled={false}
                        id="login-button"
                        onClick={[Function]}
                      >
                        Login
                      </button>
                    </form>
                  </div>
                </div>
              </div>
            </div>
          </section>
        </Login>
      </ResetOnUnmount>
    </Connect(ResetOnUnmount)>
  </Connect(Connect(ResetOnUnmount))>
</Provider>
`;

exports[`containers/login renders login in loading state 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Connect(ResetOnUnmount))>
    <Connect(ResetOnUnmount)
      isAuth={[Function]}
      isLoading={true}
      login={[Function]}
      logout={[Function]}
      password="password"
      setPassword={[Function]}
      setUsername={[Function]}
      username="username"
    >
      <ResetOnUnmount
        isAuth={[Function]}
        isLoading={true}
        login={[Function]}
        logout={[Function]}
        password="password"
        reset={[Function]}
        setPassword={[Function]}
        setUsername={[Function]}
        username="username"
      >
        <Login
          isAuth={[Function]}
          isLoading={true}
          login={[Function]}
          logout={[Function]}
          password="password"
          reset={[Function]}
          setPassword={[Function]}
          setUsername={[Function]}
          username="username"
        >
          <section
            className="hero is-fullheight"
          >
            <div
              className="hero-body"
            >
              <div
                className="container has-text-centered"
              >
                <div
                  className="column is-4 is-offset-4"
                >
                  <div
                    className="box"
                  >
                    <p
                      className="title is-1"
                    >
                      IoT
                    </p>
                    <form>
                      <div
                        className="field"
                      >
                        <div
                          className="control has-icons-left"
                        >
                          <input
                            className="input is-large"
                            id="username-input"
                            onChange={[Function]}
                            placeholder="Username"
                            value="username"
                          />
                          <span
                            className="icon is-large is-left"
                          >
                            <i
                              className="fas fa-user"
                            />
                          </span>
                        </div>
                      </div>
                      <div
                        className="field"
                      >
                        <div
                          className="control has-icons-left"
                        >
                          <input
                            className="input is-large"
                            id="password-input"
                            onChange={[Function]}
                            placeholder="Password"
                            type="password"
                            value="password"
                          />
                          <span
                            className="icon is-large is-left"
                          >
                            <i
                              className="fas fa-lock"
                            />
                          </span>
                        </div>
                      </div>
                      <button
                        className="button is-block is-primary is-large is-fullwidth is-loading"
                        disabled={true}
                        id="login-button"
                        onClick={[Function]}
                      >
                        Login
                      </button>
                    </form>
                  </div>
                </div>
              </div>
            </div>
          </section>
        </Login>
      </ResetOnUnmount>
    </Connect(ResetOnUnmount)>
  </Connect(Connect(ResetOnUnmount))>
</Provider>
`;

exports[`containers/login simulates successful login 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Connect(ResetOnUnmount))>
    <Connect(ResetOnUnmount)
      isAuth={[Function]}
      isLoading={false}
      login={[Function]}
      logout={[Function]}
      password={null}
      setPassword={[Function]}
      setUsername={[Function]}
      username={null}
    >
      <ResetOnUnmount
        isAuth={[Function]}
        isLoading={false}
        login={[Function]}
        logout={[Function]}
        password={null}
        reset={[Function]}
        setPassword={[Function]}
        setUsername={[Function]}
        username={null}
      >
        <Login
          isAuth={[Function]}
          isLoading={false}
          login={[Function]}
          logout={[Function]}
          password={null}
          reset={[Function]}
          setPassword={[Function]}
          setUsername={[Function]}
          username={null}
        >
          <section
            className="hero is-fullheight"
          >
            <div
              className="hero-body"
            >
              <div
                className="container has-text-centered"
              >
                <div
                  className="column is-4 is-offset-4"
                >
                  <div
                    className="box"
                  >
                    <p
                      className="title is-1"
                    >
                      IoT
                    </p>
                    <form>
                      <div
                        className="field"
                      >
                        <div
                          className="control has-icons-left"
                        >
                          <input
                            className="input is-large"
                            id="username-input"
                            onChange={[Function]}
                            placeholder="Username"
                            value=""
                          />
                          <span
                            className="icon is-large is-left"
                          >
                            <i
                              className="fas fa-user"
                            />
                          </span>
                        </div>
                      </div>
                      <div
                        className="field"
                      >
                        <div
                          className="control has-icons-left"
                        >
                          <input
                            className="input is-large"
                            id="password-input"
                            onChange={[Function]}
                            placeholder="Password"
                            type="password"
                            value=""
                          />
                          <span
                            className="icon is-large is-left"
                          >
                            <i
                              className="fas fa-lock"
                            />
                          </span>
                        </div>
                      </div>
                      <button
                        className="button is-block is-primary is-large is-fullwidth"
                        disabled={false}
                        id="login-button"
                        onClick={[Function]}
                      >
                        Login
                      </button>
                    </form>
                  </div>
                </div>
              </div>
            </div>
          </section>
        </Login>
      </ResetOnUnmount>
    </Connect(ResetOnUnmount)>
  </Connect(Connect(ResetOnUnmount))>
</Provider>
`;

exports[`containers/login simulates successful login 2`] = `
Object {
  "app": Object {
    "isHamburgerMenuExpanded": false,
    "isMapDialogOpened": false,
    "showError": true,
  },
  "auth": Object {
    "isAuth": false,
    "password": null,
    "username": null,
  },
  "request": Object {
    "error": null,
    "pending": 0,
    "statusCode": null,
  },
  "stats": Object {
    "filters": Object {
      "dateFilter": Object {
        "custom": Object {
          "endDate": null,
          "startDate": null,
        },
        "isCustomSelected": false,
        "timePeriod": Object {
          "isActive": false,
          "isDisabled": false,
          "isLoading": false,
          "items": Array [],
          "selectedItem": null,
        },
      },
      "items": Array [],
      "thingFilter": Object {
        "isActive": false,
        "isDisabled": false,
        "isLoading": false,
        "items": Array [],
        "selectedItem": null,
      },
      "type": Object {
        "isActive": false,
        "isDisabled": false,
        "items": Array [
          "date",
          "thing",
        ],
      },
    },
    "isLoading": false,
    "items": Array [],
    "params": Object {
      "observation": Object {
        "isActive": false,
        "isDisabled": true,
        "isLoading": false,
        "items": Array [],
        "selectedItem": null,
      },
      "reset": Object {
        "isDisabled": true,
      },
      "type": Object {
        "isActive": false,
        "items": Array [
          "event",
          "measurement",
        ],
        "selectedItem": null,
      },
    },
  },
  "things": Object {
    "items": Array [],
    "selectedItem": null,
    "shouldShowNotFoundError": false,
  },
}
`;

exports[`containers/login simulates successful login 3`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Connect(ResetOnUnmount))>
    <Connect(ResetOnUnmount)
      isAuth={[Function]}
      isLoading={false}
      login={[Function]}
      logout={[Function]}
      password={null}
      setPassword={[Function]}
      setUsername={[Function]}
      username={null}
    >
      <ResetOnUnmount
        isAuth={[Function]}
        isLoading={false}
        login={[Function]}
        logout={[Function]}
        password={null}
        reset={[Function]}
        setPassword={[Function]}
        setUsername={[Function]}
        username={null}
      >
        <Login
          isAuth={[Function]}
          isLoading={false}
          login={[Function]}
          logout={[Function]}
          password={null}
          reset={[Function]}
          setPassword={[Function]}
          setUsername={[Function]}
          username={null}
        >
          <section
            className="hero is-fullheight"
          >
            <div
              className="hero-body"
            >
              <div
                className="container has-text-centered"
              >
                <div
                  className="column is-4 is-offset-4"
                >
                  <div
                    className="box"
                  >
                    <p
                      className="title is-1"
                    >
                      IoT
                    </p>
                    <form>
                      <div
                        className="field"
                      >
                        <div
                          className="control has-icons-left"
                        >
                          <input
                            className="input is-large"
                            id="username-input"
                            onChange={[Function]}
                            placeholder="Username"
                            value=""
                          />
                          <span
                            className="icon is-large is-left"
                          >
                            <i
                              className="fas fa-user"
                            />
                          </span>
                        </div>
                      </div>
                      <div
                        className="field"
                      >
                        <div
                          className="control has-icons-left"
                        >
                          <input
                            className="input is-large"
                            id="password-input"
                            onChange={[Function]}
                            placeholder="Password"
                            type="password"
                            value=""
                          />
                          <span
                            className="icon is-large is-left"
                          >
                            <i
                              className="fas fa-lock"
                            />
                          </span>
                        </div>
                      </div>
                      <button
                        className="button is-block is-primary is-large is-fullwidth"
                        disabled={false}
                        id="login-button"
                        onClick={[Function]}
                      >
                        Login
                      </button>
                    </form>
                  </div>
                </div>
              </div>
            </div>
          </section>
        </Login>
      </ResetOnUnmount>
    </Connect(ResetOnUnmount)>
  </Connect(Connect(ResetOnUnmount))>
</Provider>
`;
